docs/guides/implement_custom_components.md

```diff
@@ -4,185 +4,28 @@ This guide will teach you how to build custom components and integrate them in y
-read a single `"image_url"` column.
-
-[//]: # (TODO: Use a transform instead of filter component here to keep it simple)
-
-Since the component only filters the data, it will not create any new data. Fondant handles your 
-data efficiently by keeping track of the index along your pipeline. Only this index will be 
-updated when filtering data, which means that we don't need to define a `produces` section in the 
-component specification.
-
-Finally, we define the arguments that the component will support. In this case, we only add a 
-single `mime_type` argument, which allows us to define which mime type should be filtered.
-
-### Implementing the component
-
-Now, it's time to implement the component logic. To do this, we'll create a `src/main.py` file.
-
-We will subclass the `PandasTransformComponent` offered by Fondant. This is the most basic type 
-of component. The following two methods should be implemented:
-
-- **`__init__()`**: This method will receive the arguments define in your component 
-  specification. Fondant also inserts some additional keyword arguments for more advanced use 
-  cases. Be sure to include a `**kwargs` argument if you're not using those.
-- **`transform()`**: This method receives a chunk of the input data as a Pandas `DataFrame`. 
-  Fondant automatically chunks your data so you can process larger-than-memory data, and your 
-  component is executed in parallel across the available cores.
-
-```python title="src/main.py"
-"""A component that filters images based on file type."""
-import mimetypes
-
-import pandas as pd
-from fondant.component import PandasTransformComponent
-
-
-class FileTypeFilter(PandasTransformComponent):
-
-    def __init__(self, *, mime_type: str, **kwargs):
-        """Custom component to filter on specific file type based on url
-        
-        Args:
-            mime_type: The mime type to filter on (also defined in the component spec)
-        """
-        self.mime_type = mime_type
-
-    @staticmethod
-    def get_mime_type(url):
-        """Guess mime type based on the file name"""
-        mime_type, _ = mimetypes.guess_type(url)
-        return mime_type
-
-    def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
-        """Reduce dataframe to specific mime type"""
-        dataframe["mime_type"] = dataframe["url"].apply(self.get_mime_type)
-        return dataframe[dataframe["mime_type"] == self.mime_type]
-```
-
-We return the filtered dataframe from the `transform` method, which Fondant will use to 
-automatically update the index. If we would have specified any output fields in our component 
-contract, Fondant would extract and write those as well.
-
-### Defining the requirements
+Reusable components consume data in a specific format, defined in a data contract.
+Therefore, it is often necessary to implement glue code to connect the reusable components to your
+specific data. The easiest way to do this is to implement a **Python component**.
```

Comment:
I would use lightweight component instead of calling it a python component which might be confusing since all components are python.